src/Nethermind/Nethermind.Blockchain/Processing/BlockchainProcessor.cs

```diff
@@ -185,24 +185,26 @@ private void RunRecoveryLoop()
             if (_logger.IsDebug) _logger.Debug($"Starting recovery loop - {_blockQueue.Count} blocks waiting in the queue.");
             foreach (BlockRef blockRef in _recoveryQueue.GetConsumingEnumerable(_loopCancellationSource.Token))
             {
-                if (!blockRef.Resolve(_blockTree))
+                if (blockRef.Resolve(_blockTree))
```

Comment:
Fliping if + FireProcessingQueueEmpty

Reply:
I remember in the past changes here were causing a lot of trouble - for actual block production, risky change, I hope that well tested

Reply:
mostly done by rider "invert if"